app/services/event.service.js

```diff
@@ -284,48 +295,48 @@
         if(filteredList.data[x].date.length > 1){
           // loop through the dates of the repeating events and pull out the object for it to loop of it with the index
           filteredList.data[x].date.forEach(function(n){
-            // Check Start Date and End Date. Only needed for All filter
-            var start = new Date(filterString.split('filter[date][value][0]=')[1].split('&')[0].split(' ')[0]).getTime() / 1000;
-            if( filterString.includes('filter[date][value][1]=') ){          
-              var end = new Date(filterString.split('filter[date][value][1]=')[1].split('&')[0].split(' ')[0]).getTime() / 1000;
-              if(n.start_unix > start && n.end_unix < end){
+            if(!data[n.start_unix]){
+              data[n.start_unix] = [];
+            }
+            // if there is an end date and started and hasn't ended
+            if( filterString.includes('filter[date][value][1]=') && n.start_unix > start && n.end_unix < end){
```

Comment:
Do we need to check for the existence of the `filter[date][value][1]=` here again, because we're already inside a loop where this was checked on line 285